pkg/httpconfig/http.go

```diff
@@ -163,7 +163,8 @@ func NewRoundTripperFromConfig(cfg config_util.HTTPClientConfig, transportConfig
 		}
 
 		if cfg.BasicAuth != nil {
-			rt = config_util.NewBasicAuthRoundTripper(cfg.BasicAuth.Username, cfg.BasicAuth.Password, cfg.BasicAuth.PasswordFile, rt)
+			// TODO(yeya24): expose UsernameFile as a config.
+			rt = config_util.NewBasicAuthRoundTripper(cfg.BasicAuth.Username, cfg.BasicAuth.Password, "", cfg.BasicAuth.PasswordFile, rt)
```

Comment:
I leave this as empty for now.
In case anyone asks for this feature, we can support and expose the config at that time.